plugins/org.eclipse.payara.tools/src/org/eclipse/payara/tools/internal/SystemLibrariesSetting.java

```diff
@@ -81,22 +91,40 @@ public static SystemLibrariesSetting load(IProject project) {
         } catch (Exception e) {
             e.printStackTrace();
         }
-        
+
         return null;
     }
 
     public static void save(IProject project, SystemLibrariesSetting settings) {
         try {
             IFile settingsXmlFile = project.getFile(SETTING_XML);
-            Marshaller settingsMarshaller = JAXBContext.newInstance(SystemLibrariesSetting.class).createMarshaller();
-            settingsMarshaller.setProperty(JAXB_FORMATTED_OUTPUT, TRUE);
-
-            // Write to System.out
-            settingsMarshaller.marshal(settings, System.out);
-
-            // Write to File
-            settingsMarshaller.marshal(settings, settingsXmlFile.getLocation().toFile());
+            XMLOutputFactory factory = XMLOutputFactory.newInstance();
+            XMLStreamWriter writer = factory.createXMLStreamWriter(
+                    new FileOutputStream(settingsXmlFile.getLocation().toFile()), StandardCharsets.UTF_8.name());
+            writer.writeStartDocument(StandardCharsets.UTF_8.name(), "1.0");
+            writer.writeCharacters(NEW_LINE);
+            writer.writeStartElement(SYSTEM_LIBRARIES_TAG);
+            writer.writeCharacters(NEW_LINE);
+            for (Library library : settings.getLibraryList()) {
+                writer.writeStartElement(LIBRARY_TAG);
+                if (library.getPath() != null) {
+                    writer.writeAttribute(PATH_TAG, library.getPath());
+                }
+                if (library.getSource() != null) {
+                    writer.writeAttribute(SOURCE_TAG, library.getSource());
+                }
+                if (library.getJavadoc() != null) {
+                    writer.writeAttribute(JAVADOC_TAG, library.getJavadoc());
+                }
+                writer.writeEndElement();
+                writer.writeCharacters(NEW_LINE);
+            }
+            writer.writeEndElement();
+            writer.writeCharacters(NEW_LINE);
+            writer.writeEndDocument();
 
+            writer.flush();
+            writer.close();
```

Comment:
Resource leak on exception - you should close the writer in finally or use trywith (you can simply create a private method XMLStreamWriter createWriter();